packages/server-core/src/user/identity-provider/identity-provider.hooks.ts

```diff
@@ -44,6 +44,18 @@ const checkIdentityProvider = (): any => {
   }
 }
 
+const checkOnlyIdentityProvider = () => {
```

Comment:
This doesn't work as intended if an identity provider is being deleted by ID; in that case, there is no query.userId, which causes a SQL query issue since it's trying to perform a find with a null value on userId. This hook should look up the provider, get the user's ID from that, and then use that in the .find() to determine how many identity-providers they have.

We actually shouldn't be supporting removing identity-providers in bulk, via query: { userId: <ID> }. I know that's how it was being done in the tests, but after thinking about it more, I don't think that's correct behavior, and the tests should be changed to delete identity-providers by ID. Currently, the only way identity-providers are removed in bulk is if a user deletes their account, and with that, we rely on SQL foreign key relationships to cascade that deletion to the identity-provider table. I don't think we ever want to delete multiple identity-providers at once.

Reply:
> This doesn't work as intended if an identity provider is being deleted by ID; in that case, there is no query.userId, which causes a SQL query issue since it's trying to perform a find with a null value on userId. This hook should look up the provider, get the user's ID from that, and then use that in the .find() to determine how many identity-providers they have.

I have added this change!

> We actually shouldn't be supporting removing identity-providers in bulk, via query: { userId: }. I know that's how it was being done in the tests, but after thinking about it more, I don't think that's correct behavior, and the tests should be changed to delete identity-providers by ID. Currently, the only way identity-providers are removed in bulk is if a user deletes their account, and with that, we rely on SQL foreign key relationships to cascade that deletion to the identity-provider table. I don't think we ever want to delete multiple identity-providers at once.

Agreed! Multiple identity providers will only be internally deleted (that too in cascade), in cases if a user is deleting his/her account. If we are going to incorporate this change, maybe we should create another PR so that this and the current issue are both issues are resolved in their separate PRs.

Reply:
Having a separate issue sounds fine.